pyproject.toml

```diff
@@ -46,7 +47,8 @@ dependencies = [
     "opentelemetry-instrumentation >= 0.41b0",
     "rich >= 13.4.2",
     "protobuf >= 4.23.4",
-    "typing-extensions >= 4.1.0",
+    'typing-extensions>=4.1.0; python_version < "3.13"',
+    'typing-extensions>=4.12.2; python_version >= "3.13"',
```

Comment:
Why is this needed?

Reply:
Becase `typing-extensions` < 4.12.2 doesn't support 3.13

Reply:
Maybe that's not true, hmm. Looking now.